native/database/db_test.go

```diff
@@ -0,0 +1,84 @@
+package database
+
+import (
+	"testing"
+)
+
+func TestInsertTransaction(t *testing.T) {
+	err := InitDB(":memory:") // in-memory database
+	if err != nil {
+		t.Fatal(err)
+	}
```

Comment:
this check should be in `initDB`.
Also use https://pkg.go.dev/gotest.tools/assert rather than "raw" methods of `t`

Reply:
Done